wolfssl/wolfcrypt/wc_port.h

```diff
@@ -219,8 +219,6 @@
         typedef CRITICAL_SECTION wolfSSL_Mutex;
     #elif defined(WOLFSSL_PTHREADS)
         typedef pthread_mutex_t wolfSSL_Mutex;
-    #elif defined(WOLFSSL_KTHREADS)
```

Comment:
How does removal of this not cause an `#error` in the else case? Is it because `WOLFSSL_LINUXKM ` is always defined when `WOLFSSL_KTHREADS` is defined? Perhaps consider changing the `#elif defined(WOLFSSL_LINUXKM)` line below to include `|| defined(WOLFSSL_KTHREADS)`?

Reply:
`WOLFSSL_KTHREADS` is defined in linuxkm_wc_port.h, and linuxkm_wc_port.h is included iff `WOLFSSL_LINUXKM` is defined.  So they track each other.  I see there's a redundant definition of `WOLFSSL_KTHREADS` and a frivolous include in internal.h.  There's only one substantive use of the `WOLFSSL_KTHREADS` macro in the codebase, in wc_port.c where the mutex wrappers are defined, and that can equivalently be gated on `WOLFSSL_LINUXKM`.

I think the best approach is to eliminate the `WOLFSSL_KTHREADS` macro, for clarity+simplicity -- makes sense to you too?

Reply:
I agree. Sounds good to eliminate WOLFSSL_KTHREADS.